File: Haskell/problem2.hs
{- (*) Find the last but one element of a list. -}

lbo :: [a] -> a
lbo [x, _] = x
lbo (_:xs) = lbo xs

lbo' :: [a] -> a
lbo' = head . tail . reverse

lbo'' :: [a] -> a
lbo'' = head . drop 1 . reverse

File: Prolog/problem2.pl
%% 1.02 (*) Find the last but one element of a list.

% last_but_one(+Xs, ?LBO)
last_but_one([X,_], X) :- !.% green cut
last_but_one([_|Xs], LBO) :- last_but_one(Xs, LBO).
